animation_nodes/nodes/object/lamp_output.py

```diff
@@ -25,5 +25,16 @@ def create(self):
             ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
             ("Objects", "objects")))
 
+        for socket in self.inputs[1:]:
+            socket.useIsUsedProperty = True
+            socket.isUsed = False
+
     def getExecutionCode(self, required):
-        return "if object is not None and object.type == 'LIGHT': object.data.color = color[:3]; object.data.energy = energy"
+        s = self.inputs
+        isColor = s[1].isUsed
+        isEnergy = s[2].isUsed
```

Comment:
Remove the s variable and use its value directly. I think a more descriptive name is useColor and useEnergy.

Reply:
Okay.

Reply:
Done.